src/etc/inc/filter.inc

```diff
@@ -483,6 +483,23 @@ function filter_configure_sync($delete_states_if_needed = true) {
 		$_grbg = exec("/sbin/pfctl -t bogonsv6 -T kill 2>/dev/null");
 	}
 
+	global $reserved_table_names;
+	// get aliases names from config
+	$aliasesnames = array();
+	foreach ($config['aliases']['alias'] as $alias) {
+		array_push($aliasesnames, $alias['name']);
+	}
+	// get aliases names from pfctl -sT
+	exec("/sbin/pfctl -sT", $pftables);
+	// compare config and pfctl aliases
+	$diff = array_diff($pftables, $aliasesnames);
+	// don't delete special aliases
+	$diff = array_diff($diff, $reserved_table_names);
```

Comment:
Prototype of array_diff `array_diff(array $array1 , array $array2 [, array $... ] )` allows to get many arrays, it will compare `$array1` with other arrays and will identify items present on `$array1` which are not present on any other array.  It seems to be the case here, so you could reduce it to a single call to `array_diff($pftables, $aliasesnames, $reserved_table_names);`

Reply:
done